include/ginkgo/core/base/utils_helper.hpp

```diff
@@ -251,9 +276,10 @@ inline typename std::remove_reference<OwningPointer>::type&& give(
  *       same as calling .get() on the smart pointer.
  */
 template <typename Pointer>
-inline typename std::enable_if<detail::have_ownership_s<Pointer>::value,
-                               detail::pointee<Pointer>*>::type
-lend(const Pointer& p)
+[[deprecated("no longer necessary due to pointer_param")]] inline
```

Comment:
Perhaps focus on what to change instead:
```suggestion
[[deprecated("no longer necessary, just pass the object without lend")]] inline
```